src/app/pages/OpenWalletPage/Features/ListBleLedgerDevicesModal/index.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { useDispatch, useSelector } from 'react-redux'
+import { AlertBox } from 'app/components/AlertBox'
+import { ErrorFormatter } from 'app/components/ErrorFormatter'
+import { ModalHeader } from 'app/components/Header'
+import { ImportAccountsStepFormatter } from 'app/components/ImportAccountsStepFormatter'
+import { ResponsiveLayer } from 'app/components/ResponsiveLayer'
+import {
+  selectBleDevices,
+  selectError,
+  selectImportAccountsStep,
+  selectSelectedBleDevice,
+} from 'app/state/importaccounts/selectors'
+import { ImportAccountsStep } from 'app/state/importaccounts/types'
+import { Box } from 'grommet/es6/components/Box'
+import { Button } from 'grommet/es6/components/Button'
+import { Spinner } from 'grommet/es6/components/Spinner'
+import { Text } from 'grommet/es6/components/Text'
+import { ScanResult } from '@capacitor-community/bluetooth-le'
+import { CheckBox } from 'grommet/es6/components/CheckBox'
+import { importAccountsActions } from '../../../../state/importaccounts'
+
+interface ImportAccountsSelectorSelectorProps {
+  devices: ScanResult[]
+}
+
+function DevicesSelector({ devices }: ImportAccountsSelectorSelectorProps) {
+  const dispatch = useDispatch()
+  const selectedBleDevice = useSelector(selectSelectedBleDevice)
+  const selectBleDevice = (scanResult: ScanResult) =>
+    dispatch(importAccountsActions.setSelectedBleDevice(scanResult))
+
+  return (
+    <Box gap="small">
+      {devices.map(d => (
+        <Box
+          key={d.device.deviceId}
+          round="5px"
+          pad="small"
+          flex="grow"
+          fill="horizontal"
+          role="checkbox"
+          hoverIndicator={{ background: 'brand' }}
+          direction="row"
+          onClick={() => {
+            selectBleDevice(d)
+          }}
+        >
+          <Box alignSelf="center" pad={{ left: 'small', right: 'medium' }}>
+            <CheckBox checked={selectedBleDevice?.device.deviceId === d.device.deviceId} hidden />
+          </Box>
+          <Box flex="grow" gap="xsmall">
+            <Box>
+              <Text weight="bold">{d.device.name}</Text>
+            </Box>
+          </Box>
+        </Box>
+      ))}
+    </Box>
+  )
+}
+
+interface ListBleLedgerDevicesModalProps {
+  abort: () => void
+  next: () => void
+}
+
+export function ListBleLedgerDevicesModal(props: ListBleLedgerDevicesModalProps) {
```

Comment:
so the difference is that we use native modal for usb and custom one for ble?

Reply:
Yes, sort of. Listing is not native, connect to a single device is. This will probably be reworked for desktop support - but the API is still not officially released yet(at least the last time I checked).